wandb/sdk/launch/builder/kaniko_builder.py

```diff
@@ -255,22 +257,35 @@ def build_image(
         _, api_client = get_kube_context_and_api_client(
             kubernetes, launch_project.resource_args
         )
+        # TODO: use same client as kuberentes.py
+        batch_v1 = client.BatchV1Api(api_client)
+        core_v1 = client.CoreV1Api(api_client)
+
         build_job_name = f"{self.build_job_name}-{run_id}"
 
         build_context = self._upload_build_context(run_id, context_path)
         build_job = self._create_kaniko_job(
-            build_job_name,
-            repo_uri,
-            image_uri,
-            build_context,
+            build_job_name, repo_uri, image_uri, build_context, core_v1
         )
         wandb.termlog(f"{LOG_PREFIX}Created kaniko job {build_job_name}")
 
-        # TODO: use same client as kuberentes.py
-        batch_v1 = client.BatchV1Api(api_client)
-        core_v1 = client.CoreV1Api(api_client)
-
         try:
+            if isinstance(self.registry, AzureContainerRegistry):
```

Comment:
I think we can make this a little more self documenting by creating a helper function (potentially a method on the registry class) to create the config map

Reply:
It would also be good to explain why we need this special config map specifically for Azure with a comment